### ISSUE TITLE
[BUG] False-positive test result for "An onscreen element must not have a null BoundingRectangle property".

### ISSUE THREAD

**Issue by giladnoy** [microsoft/axe-windows#387]
## Description

We want to implement a scrollable list in our app, which has complex items (e.g. a group containing a list).
When one of the complex items is scrolled out of visible part of the app, and the Accessibility Insights tool test is being run, the inner component of the complex list item is falsely considered as having the property "IsOffScreen" set to true, although its parent does have the correct value for this property (false).
**Note**: This scenario was also seen when running the Accessibility Insights tool on the Windows BT devices settings page.

Similar issues:
https://github.com/microsoft/axe-windows/issues/357
https://github.com/microsoft/axe-windows/issues/352

**To Reproduce**
Precondition: Have a non-empty list in the "Other Devices" section on the Windows BT settings page.
1. Go to Windows -> Settings
2. Click on Devices -> Bluetooth & other devices
3. Scroll all the way to the top, and make sure the "Other Devices" section is scrolled out of the app.
2. Run Accessibility Insights test on the Settings window.

## Expected behavior

All tests should pass

## Actual behavior
The "Other Devices" group, which is scrolled off screen, has the IsOffScreen property is set to **true**, while one of its children (a list which is also offscreen) has the IsOfScreen property set to **false**.
This causes the test to fail with the error: "**An onscreen element must not have a null BoundingRectangle property**."

## Screenshots
<img width="960" alt="List inside the Other Devices group" src="https://user-images.githubusercontent.com/31479778/85225760-226d8000-b3dc-11ea-85a9-6d78f959f9f8.png">
<img width="955" alt="Other devices group" src="https://user-images.githubusercontent.com/31479778/85225764-24374380-b3dc-11ea-9e3a-ceec96e2b722.png">

## Desktop

 - OS: Windows 10 1909
 - Accessibility Insights for Windows Version: 1.1.1248.1
 - Target Application: UWP

**Comment by RobGallo** [comment]
Hi, @giladnoy 
Thanks for the detailed bug report! Would you mind sending us an a11ytest file? You can generate this by reproducing the problem (running the scan) in AI-Win, then choosing the "save" button at the top of the screen. That would help us to analyze the issue. You can attach the file to a comment in this GH issue. Thanks!

**Comment by giladnoy** [comment]
Yeah, sure:)
[TestResults Windows BT devices 06-21-20.zip](https://github.com/microsoft/axe-windows/files/4814546/TestResults.Windows.BT.devices.06-21-20.zip)

**Comment by RobGallo** [comment]
Thanks. Looking in to this.

**Comment by RobGallo** [comment]
We believe this issue is not a false positive because the data should either reflect that the list is off screen and therefore the bounding rectangle may be null, or that the list is on screen and the bounding rectangle must be provided.

@giladnoy , This is a [known bug in xaml](https://github.com/microsoft/microsoft-ui-xaml/issues/1979) which you can reference for accessibility reporting purposes.

**Comment by giladnoy** [comment]
@RobGallo If we were discussing this element by itself, I would completely agree.
However, according to MS documentation regarding **IsOffScreen** property (https://docs.microsoft.com/en-us/dotnet/api/system.windows.automation.automationelement.automationelementinformation.isoffscreen?view=netcore-3.1):

> When the value is **true** for a container, it is also **true** for the container element's descendants.

Doesn't that mean that if the **IsOffScreen** property of some element is true, we can safely assume all of its descendants have the same value for that property, and not present an error when some descendant has a false value?

**Comment by RobGallo** [comment]
@giladnoy ,
I agree the language there is confusing. But we think that text is targetted toward the app developer, not the assistive technology developer. So we take the meaning of the paragraph you quoted to be when an element's isOffScreen property is set to true, all of the descendants of that element should also have their isOffScreen property set to true. We even had a rule to enforce the structure, but we had to remove it because it was violated so often and only framework developers could fix it.
The reason we interpretted the text in this way is because assistive technologies which consume UIA should not have to query the ancestry of every element to calculate whether or not it is off screen because that action is expensive. Far better to have each element to reflect its correct isOffScreen state.
property
